homeassistant/components/sensor/azure_maps_travel_time.py

```diff
@@ -0,0 +1,246 @@
+
+    def __init__(self, hass, name, api_key, origin, destination, options):
+        """Initialize the sensor."""
+        self._hass = hass
+        self._key = api_key
+        self._name = name
+        self._options = options
+        self._unit_of_measurement = 'min'
+        self._state = None
+        self._attributes = None
+        self.valid_api_connection = True
+
+        # Check if location is a trackable entity
+        if origin.split('.', 1)[0] in TRACKABLE_DOMAINS:
+            self._origin_entity_id = origin
+        else:
+            self._origin = origin
+
+        if destination.split('.', 1)[0] in TRACKABLE_DOMAINS:
+            self._destination_entity_id = destination
+        else:
+            self._destination = destination
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def name(self):
+        """Get the name of the sensor."""
+        return self._name
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return self._attributes
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit this state is expressed in."""
+        return self._unit_of_measurement
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def update(self):
+        """Get the latest data from Azure Maps."""
+        options_copy = dict(self._options.copy())
+        dtime = options_copy.get('departAt')
+        atime = options_copy.get('arriveAt')
+        if dtime is not None and ':' in dtime:
+            options_copy['departAt'] = convert_time_to_utc(dtime).isoformat()
+        elif dtime is not None:
+            options_copy['departAt'] = dtime.isoformat()
+        if atime is not None and ':' in atime:
+            options_copy['arriveAt'] = convert_time_to_utc(atime).isoformat()
+        elif atime is not None:
+            options_copy['arriveAt'] = atime.isoformat()
+
+        # Convert device_trackers to friendly locations
+        if hasattr(self, '_origin_entity_id'):
+            self._origin = self._get_location_from_entity(
+                self._origin_entity_id
+            )
+        if hasattr(self, '_destination_entity_id'):
+            self._destination = self._get_location_from_entity(
+                self._destination_entity_id
+            )
+
+        self._destination = self._resolve_zone(self._destination)
+        self._origin = self._resolve_zone(self._origin)
+
+        if self._destination is not None and self._origin is not None:
+            _LOGGER.debug(
+                'origin: '+self._origin+', destination: '+self._destination)
+            options_copy['query'] = self._origin + ":" + self._destination
+            options_copy['api-version'] = 1.0
+            options_copy['subscription-key'] = self._key
+            options_copy['traffic'] = True
+            options_copy['ComputeTravelTimeFor'] = 'all'
+
+            resp = requests.get(
```

Comment:
This needs to be extracted into a PyPi package as per our guidelines:

https://developers.home-assistant.io/docs/en/creating_platform_code_review.html#6-communication-with-devices-services

Yes, this code needs to be in a PyPi package too, even if it's just a single call. No discussion.

Reply:
ok, I'll work on that!